Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserEntityInspectorWidget.cpp

```diff
@@ -25,6 +27,90 @@ namespace AzToolsFramework
 {
     namespace AssetBrowser
     {
+        class DependentAssetTreeWidgetItem : public QTreeWidgetItem
+        {
+        public:
+            DependentAssetTreeWidgetItem(const AssetBrowserEntry* assetBrowserEntry, QTreeWidgetItem* parent)
+                : m_assetBrowserEntry(assetBrowserEntry)
+                , QTreeWidgetItem(parent)
+            {}
+
+            const AssetBrowserEntry* m_assetBrowserEntry;
+        };
+
+        class DependentAssetTreeWidget : public QTreeWidget
+        {
+        public:
+            explicit DependentAssetTreeWidget(QWidget* parent = nullptr)
+                : QTreeWidget(parent)
+            {
+                setContextMenuPolicy(Qt::CustomContextMenu);
+
+                connect(this, &QTreeWidget::customContextMenuRequested, this, &DependentAssetTreeWidget::OnContextMenu);
+            }
+
+            ~DependentAssetTreeWidget() override = default;
+
+        private:
+            void mouseDoubleClickEvent([[maybe_unused]] QMouseEvent* ev) override
+            {
+                if (const DependentAssetTreeWidgetItem* widgetItem = GetSelectedWidgetItem())
+                {
+                    BrowseToAsset(*widgetItem);
+                }
+            }
+
+            void OnContextMenu(const QPoint& point)
+            {
+                const DependentAssetTreeWidgetItem* widgetItem = GetSelectedWidgetItem();
+                if (!widgetItem)
+                {
+                    return;
+                }
+
+                QMenu menu(this);
+                menu.addAction(
+                    QObject::tr("Browse to asset"),
+                    [widgetItem]()
+                    {
+                        BrowseToAsset(*widgetItem);
+                    });
+                menu.exec(mapToGlobal(point));
+            }
+
+            const DependentAssetTreeWidgetItem* GetSelectedWidgetItem() const
+            {
+                const QModelIndexList selection = selectedIndexes();
+                if (selection.isEmpty())
+                {
+                    return nullptr;
+                }
+
+                const auto* item = static_cast<QTreeWidgetItem*>(selection.first().internalPointer());
+                return dynamic_cast<const DependentAssetTreeWidgetItem*>(item);
```

Comment:
We should avoid using `dynamic_cast`. Since this a Qt derived class, try using [qobject_cast](https://doc.qt.io/qt-6/qobject.html#qobject_cast-1) instead

Reply:
I wasn't a fan of exposing the DependentAssetTreeWidgetItem in the .h (needed for the qobject_cast) so instead I made all items of the tree using the same type so that casting will never fail (removed the dynamic cast). Let me know if you have issues with this approach :)